kinesis/src/main/scala/akka/stream/alpakka/kinesis/scaladsl/KinesisFlow.scala

```diff
@@ -39,41 +48,48 @@ object KinesisFlow {
                 getPayloadByteSize,
                 ThrottleMode.Shaping)
       .batch(settings.maxBatchSize, Queue(_))(_ :+ _)
-      .via(
-        new KinesisFlowStage(
-          streamName,
-          settings.maxRetries,
-          settings.backoffStrategy,
-          settings.retryInitialTimeout
-        )
+      .mapAsync(settings.parallelism)(
+        entries =>
+          kinesisClient
+            .putRecords(
+              PutRecordsRequest.builder().streamName(streamName).records(entries.map(_._1).asJavaCollection).build
+            )
+            .toScala
+            .transform(handlePutRecordsSuccess(entries), FailurePublishingRecords(_))(sameThreadExecutionContext)
       )
-      .mapAsync(settings.parallelism)(identity)
       .mapConcat(identity)
 
+  private def handlePutRecordsSuccess[T](
+      entries: Iterable[(PutRecordsRequestEntry, T)]
+  )(result: PutRecordsResponse): List[(PutRecordsResultEntry, T)] =
+    result.records.asScala.toList.zip(entries).map { case (res, (_, t)) => (res, t) }
```

Comment:
Hm, this unrolls the `PutRecordsResponse` so the user can't actually react on a non-zero `failedRecordCount`. I guess the records with errors will be missing in the response, that's why the old implementation had some correlation logic.
Ideas?

Reply:
The `PutRecordsResultEntry` is enough to determine whether an error occurred at the record level (via `errorCode`), isn't it?

Reply:
Ok, I didn't discover that API.
The examples should show something along the lines of
```
  .map { response =>
    if (response.errorCode != 0) throw new RuntimeException(...)
    response
  }
```
so that users understand it is their responsibility to check for response errors.